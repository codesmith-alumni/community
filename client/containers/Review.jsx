import React from "react";
import { connect } from "react-redux";
import styled from "styled-components";

const ReviewStyles = styled.div`
  background-color: white;
  height: 150px;
  width: 100%;
  border: solid;

`;


class Review extends React.Component {
  constructor(props){
    super(props)

  }



  render() {

    return (
        <ReviewStyles>
         Company: {this.props.company}
         Username: {this.props.username}
         Details: {this.props.details}
        </ReviewStyles>

    );
  }
}

export default Review;