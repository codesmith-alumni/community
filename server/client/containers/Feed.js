"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = require("react");
var styled_components_1 = require("styled-components");
var Review_jsx_1 = require("../containers/Review.jsx");
var FeedStyles = styled_components_1.default.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  height: 550px;\n  width: 100%;\n  border: solid;\n\n\n"], ["\n  height: 550px;\n  width: 100%;\n  border: solid;\n\n\n"])));
var Feed = /** @class */ (function (_super) {
    __extends(Feed, _super);
    function Feed(props) {
        return _super.call(this, props) || this;
    }
    Feed.prototype.render = function () {
        var contents = [];
        var posts = this.props.posts;
        posts.forEach(function (item, index) {
            contents.push(<Review_jsx_1.default {...item} key={index}/>);
        });
        return (<FeedStyles>
          {contents}
        </FeedStyles>);
    };
    return Feed;
}(react_1.default.Component));
exports.default = Feed;
var templateObject_1;
// const Feed = ({className, posts}) => {
//   const [posts, setPosts] = useState('')
//   const contents = []
//   // const {posts} = this.props
//   posts.forEach((item, index) => {
//     contents.push(<Review {...item} key = {index}/>)
//     setPosts(contents)
//   })
//   return(
//     <FeedStyles>
//      {contents}
//     </FeedStyles>
//   )
// }
//# sourceMappingURL=Feed.js.map