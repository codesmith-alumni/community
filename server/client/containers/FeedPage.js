"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = require("react");
var react_redux_1 = require("react-redux");
var styled_components_1 = require("styled-components");
var actions_1 = require("../actions/actions");
var Search_jsx_1 = require("../containers/Search.jsx");
var Feed_jsx_1 = require("../containers/Feed.jsx");
var Composer_jsx_1 = require("./Composer.jsx");
var FeedPageStyles = styled_components_1.default.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  background-color: lightskyblue;\n  height: 100vh;\n  width: 100vw;\n  \n"], ["\n  background-color: lightskyblue;\n  height: 100vh;\n  width: 100vw;\n  \n"])));
var mapStateToProps = function (store) { return ({
    posts: store.posts,
}); };
var mapDispatchToProps = function (dispatch) { return ({
    togglePost: function (index) { return dispatch(actions_1.toggle(index)); },
    updatePosts: function (posts) { return dispatch(actions_1.updatePosts(posts)); },
}); };
var FeedPage = /** @class */ (function (_super) {
    __extends(FeedPage, _super);
    function FeedPage(props) {
        var _this = _super.call(this, props) || this;
        _this.state = {};
        _this.handleSearch = _this.handleSearch.bind(_this);
        _this.handleReview = _this.handleReview.bind(_this);
        return _this;
    }
    FeedPage.prototype.handleSearch = function (searchTerm) {
        console.log('handleSearch');
        this.getPosts(searchTerm);
    };
    FeedPage.prototype.getPosts = function (searchTerm) {
        var _this = this;
        var path = searchTerm ? "/posts/" + searchTerm : "/posts/";
        fetch(path)
            .then(function (response) { return response.json(); })
            .then(function (response) {
            var posts = [];
            response.forEach(function (item) {
                posts.push({
                    isOpen: false,
                    company: item.company,
                    details: item.content,
                    username: item.user_id
                });
            });
            return _this.props.updatePosts(posts);
        })
            .catch(function (err) { return console.log(err); });
    };
    FeedPage.prototype.handleReview = function (reviewCompany, reviewText) {
        console.log('handleReview');
        this.sendReview(reviewCompany, reviewText);
    };
    //Need to include user_id in body in fetch request below -> body: JSON.stringify()
    FeedPage.prototype.sendReview = function (reviewCompany, reviewText, getPosts) {
        console.log(reviewCompany, reviewText);
        var path = reviewCompany && reviewText ? "/posts" : "/posts/";
        fetch(path, { method: 'POST', body: JSON.stringify({ company: reviewCompany, content: reviewText }) })
            .then(getPosts)
            .catch(function (err) { return console.log(err); });
    };
    FeedPage.prototype.handleSearch = function (searchTerm) {
        console.log('handleSearch');
        this.getPosts(searchTerm);
    };
    FeedPage.prototype.componentDidMount = function () {
        this.getPosts();
    };
    FeedPage.prototype.render = function () {
        return (<FeedPageStyles>
          Stream
        <Search_jsx_1.default handleSearch={this.handleSearch}/>
        <Composer_jsx_1.default handleReview={this.handleReview}/>
        <Feed_jsx_1.default posts={this.props.posts}/>
        
        </FeedPageStyles>);
    };
    return FeedPage;
}(react_1.default.Component));
exports.default = react_redux_1.connect(mapStateToProps, mapDispatchToProps)(FeedPage);
var templateObject_1;
//# sourceMappingURL=FeedPage.js.map