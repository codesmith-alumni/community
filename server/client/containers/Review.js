"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = require("react");
var styled_components_1 = require("styled-components");
var ReviewStyles = styled_components_1.default.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  background-color: white;\n  height: 50px;\n  width: 100%;\n  border: solid;\n\n"], ["\n  background-color: white;\n  height: 50px;\n  width: 100%;\n  border: solid;\n\n"])));
var Review = /** @class */ (function (_super) {
    __extends(Review, _super);
    function Review(props) {
        return _super.call(this, props) || this;
    }
    Review.prototype.render = function () {
        return (<ReviewStyles>
         Company: {this.props.company}
         Username: {this.props.username}
         Details: {this.props.details}
        </ReviewStyles>);
    };
    return Review;
}(react_1.default.Component));
exports.default = Review;
var templateObject_1;
//# sourceMappingURL=Review.js.map