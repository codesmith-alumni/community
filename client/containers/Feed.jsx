import React, {useState} from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import Review from "../containers/Review.jsx"

const FeedStyles = styled.div`
  height: 550px;
  width: 100%;
  border: solid;


`;


class Feed extends React.Component {
constructor(props){
  super(props)
}
render() {
  const contents = []
  const {posts} = this.props
    posts.forEach((item, index) => {
      contents.push(<Review {...item} key = {index}/>)
    })
    return (
      <FeedStyles>
          {contents}
        </FeedStyles>
    );
  }
}

export default Feed;




  // const Feed = ({className, posts}) => {
  //   const [posts, setPosts] = useState('')
  //   const contents = []
  //   // const {posts} = this.props
  //   posts.forEach((item, index) => {
  //     contents.push(<Review {...item} key = {index}/>)
  //     setPosts(contents)
  //   })
  
  //   return(
  //     <FeedStyles>
  //      {contents}
  //     </FeedStyles>
  
  
  //   )
  
  // }