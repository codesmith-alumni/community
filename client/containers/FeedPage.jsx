import React from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import { toggle, updatePosts } from "../actions/actions";
import   Search   from "../containers/Search.jsx"
import Feed from "../containers/Feed.jsx"
import Composer from "../containers/Composer.jsx"

const FeedPageStyles = styled.div`
  background-color: lightcoral;
  height: 150px;
  width: 100%;
  border: solid;

`;

const mapStateToProps = store => ({
  posts: store.posts
});

const mapDispatchToProps = dispatch => ({
  togglePost: index => dispatch(toggle(index)),
  updatePosts: posts => dispatch(updatePosts(posts))
});

class FeedPage extends React.Component {
  constructor(props){
    super(props)
this.state = {}
this.handleSearch = this.handleSearch.bind(this)

  } 

getPosts(searchTerm){
  const path = searchTerm ? `/posts/${searchTerm}` : `/posts/`
  
  fetch(path)
  .then(response => response.json())
  .then(response => {
    const posts = []
  response.forEach(item => {
    posts.push({
      isOpen: false,
      company: item.company,
      details:item.content,
      username:item.user_id

    })
  })
   return this.props.updatePosts(posts)})
  .catch(err => console.log(err))
}


 handleSearch(searchTerm){
  console.log('handleSearch')
  this.getPosts(searchTerm)

  }



  componentDidMount(){

   this.getPosts()

  }
 

  render() {

    return (
        <FeedPageStyles>
          Feed I am the feed page.
        <Search handleSearch = {this.handleSearch}/>
        <Composer/>
        <Feed posts = {this.props.posts}/>
          <h1>value of toggled: {this.props.posts[0].isOpen.toString()}</h1>
          <button onClick={index => this.props.togglePost(0)}> toggle</button>
        </FeedPageStyles>

    );
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FeedPage);
