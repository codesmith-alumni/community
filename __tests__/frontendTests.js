import React from 'react';
import renderer from 'react-test-renderer'
import FeedPage from '../client/containers/FeedPage'
import Review from '../client/containers/Review'
import Feed from '../client/containers/Feed'




describe('Review tests', () => {
  it('render correctly', () => {
    const props = {company: 'Google', username: 'Camera', details: 'Crushed it!'}
    const component = renderer.create(<Review {...props}/>)
    .toJSON()
    expect(component).toMatchSnapshot()
  }) 
  
})

describe('Feed tests', () => {
  it('render correctly', () => {
    const posts = [{company: 'Google', username: 'Camera', details: 'Crushed it!'}]
    const component = renderer.create(<Feed posts = {posts} />)
    .toJSON()
    expect(component).toMatchSnapshot()
  })

})
