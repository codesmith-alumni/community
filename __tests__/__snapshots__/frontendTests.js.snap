// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed tests render correctly 1`] = `
<div
  className="Feed__FeedStyles-sc-132b3eh-0 fXojgU"
>
  <div
    className="Review__ReviewStyles-ga6o6v-0 hUmNth"
  >
    Company: 
    Google
    Username: 
    Camera
    Details: 
    Crushed it!
  </div>
  I am the Feed
</div>
`;

exports[`Review tests render correctly 1`] = `
<div
  className="Review__ReviewStyles-ga6o6v-0 hUmNth"
>
  Company: 
  Google
  Username: 
  Camera
  Details: 
  Crushed it!
</div>
`;
